ts-client/src/example.ts

```diff
@@ -109,38 +199,136 @@ async function getPositionsState(dlmmPool: DLMM) {
 }
 
 async function addLiquidityToExistingPosition(dlmmPool: DLMM) {
+  const TOTAL_RANGE_INTERVAL = 10; // 10 bins on each side of the active bin
+  const bins = [activeBin.binId]; // Make sure bins is less than 70, as currently only support up to 70 bins for 1 position
+  for (
+    let i = activeBin.binId;
+    i < activeBin.binId + TOTAL_RANGE_INTERVAL / 2;
+    i++
+  ) {
+    const rightNextBinId = i + 1;
+    const leftPrevBinId = activeBin.binId - (rightNextBinId - activeBin.binId);
+    bins.push(rightNextBinId);
+    bins.unshift(leftPrevBinId);
+  }
+
+  const activeBinPricePerToken = dlmmPool.fromPricePerLamport(
+    Number(activeBin.price)
+  );
+  const totalXAmount = new BN(100);
+  const totalYAmount = totalXAmount.mul(new BN(Number(activeBinPricePerToken)));
+
   // Add Liquidity to existing position
-  const addLiquidityTx = await dlmmPool.addLiquidityByWeight({
-    positionPubKey: userPositions[0].publicKey,
+  const addLiquidityTx = await dlmmPool.addLiquidityByStrategy({
+    positionPubKey: newBalancePosition.publicKey,
     user: user.publicKey,
     totalXAmount,
     totalYAmount,
-    xYAmountDistribution: spotXYAmountDistribution,
+    strategy: {
+      maxBinId: bins[bins.length - 1],
+      minBinId: bins[0],
+      strategyType: StrategyType.SpotBalanced,
+    },
+  });
+
+  try {
+    const addLiquidityTxHash = await sendAndConfirmTransaction(
+      connection,
+      addLiquidityTx,
+      [user]
+    );
+    console.log("🚀 ~ addLiquidityTxHash:", addLiquidityTxHash);
+  } catch (error) {
+    console.log("🚀 ~ error:", JSON.parse(JSON.stringify(error)));
+  }
+}
+
+async function removeBalancePositionLiquidity(dlmmPool: DLMM) {
```

Comment:
I think a single`removePositionLiquidity` function is sufficient, because as the market price move, it might be consists of both, or single token.